### PR TITLE
Dockerfile: stop doing useless stuff and do others properly

### PR DESCRIPTION
**Pull request by ignatenkobrain**
Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
Doing echo which is copying commands itself is worst idea ever. This is just monkey copy-pasing thing. There is special mode in shell which will enable verbose mod (which will print executed commands as well).

File: Dockerfile [hunk 1]
```diff
@@ -7,46 +7,38 @@ COPY dnf-docker-test/features /behave/
 COPY dnf-docker-test/features /tests/
 COPY rpms /rpms/
 
-RUN echo -e '\necho "deltarpm=0" >> /etc/dnf/dnf.conf' \
+RUN set -x && \
```
Comment (j-mracek) [inline Dockerfile]: I guess this will not work && \ &&

Comment (ignatenkobrain) [inline Dockerfile]: fixed

File: Dockerfile [hunk 2]
```diff
@@ -7,46 +7,38 @@ COPY dnf-docker-test/features /behave/
 COPY dnf-docker-test/features /tests/
 COPY rpms /rpms/
 
-RUN echo -e '\necho "deltarpm=0" >> /etc/dnf/dnf.conf' \
+RUN set -x \
```
Comment (j-mracek) [inline Dockerfile]: In my terminal it pollute output with staff like this:
```
++ __vte_prompt_command
+++ HISTTIMEFORMAT=
+++ history 1
+++ sed 's/^ *[0-9]\+ *//'
++ local 'command=rm -fv aqdasdsd'
++ command='rm -fv aqdasdsd'
++ local 'pwd=~'
++ '[' /home/jmracek/Projects/libhif '!=' /home/jmracek ']'
++ pwd='~/Projects/libhif'
+++ __vte_osc7
++++ __vte_urlencode /home/jmracek/Projects/libhif
++++ LC_ALL=C
++++ str=/home/jmracek/Projects/libhif
++++ '[' -n /home/jmracek/Projects/libhif ']'
++++ safe=/home/jmracek/Projects/libhif
++++ printf %s /home/jmracek/Projects/libhif
++++ str=
++++ '[' -n '' ']'
++++ '[' -n '' ']'
+++ printf '\033]7;file://%s%s\007' dhcp-0-213.brq.redhat.com /home/jmracek/Projects/libhif
++ printf '\033]777;notify;Command completed;%s\007\033]0;%s@%s:%s\007%s' 'rm -fv aqdasdsd' jmracek dhcp-0-213 '~/Projects/libhif' ''
```
Please remove it.

File: Dockerfile [hunk 3]
```diff
@@ -7,46 +7,38 @@ COPY dnf-docker-test/features /behave/
...
+    # update dnf
     && dnf -y --best upgrade dnf \
-    && echo -e '\nif ls /rpms/*.rpm 1> /dev/null 2>&1; then dnf -y install /rpms/*.rpm; else echo "RPMs files do not exist" >&2; fi' \
-    && if ls /rpms/*.rpm 1>/dev/null 2>&1; then dnf -y install /rpms/*.rpm else echo "RPMs files do not exist" >&2; fi \
     && if [ $type = "local" ] ; then \
-        # next line is a temporary workaround for RhBug:1398272
-        rpm -q dnf \
-        && echo -e '\ndnf -y mark install $(dnf repoquery --unneeded -q)' \
+        # install all rpms if present
+        if ls /rpms/*.rpm 1>/dev/null 2>&1; then dnf -y install /rpms/*.rpm; fi \
```
Comment (j-mracek) [inline Dockerfile]: add echo to make visible whats going on. Important for debugging.

File: Dockerfile [hunk 4]
```diff
@@ -7,46 +7,38 @@ COPY dnf-docker-test/features /behave/
...
-        echo -e '\ndnf -y install dnf-plugins-core python3-dnf-plugins-core python2-dnf-plugins-core rpm-build createrepo_c' \
         && dnf -y install dnf-plugins-core python3-dnf-plugins-core python2-dnf-plugins-core rpm-build createrepo_c  \
-        && echo -e '\ndnf -y copr enable rpmsoftwaremanagement/dnf-nightly' \
         && dnf -y copr enable rpmsoftwaremanagement/dnf-nightly ; \
     fi \
     # prevent installation of dnf-plugins-extras (versionlock, local, torproxy)
-    && echo -e '\nrm /rpms/*extras-versionlock*.rpm /rpms/*extras-local*.rpm /rpms/*extras-torproxy*.rpm' \
-    && rm /rpms/*extras-versionlock*.rpm /rpms/*extras-local*.rpm /rpms/*extras-torproxy*.rpm || echo "RPMs files do not exist" >&2 \
-    && echo -e '\ndnf -y --best upgrade dnf' \
+    && rm -vf /rpms/*extras-versionlock*.rpm /rpms/*extras-local*.rpm /rpms/*extras-torproxy*.rpm \
```
Comment (j-mracek) [inline Dockerfile]: add echo to make visible whats going on. Important for debugging.

File: Dockerfile (hunk 2 above)
Comment (ignatenkobrain) [inline Dockerfile]: because your terminal != docker build environment.

File: Dockerfile (hunk 3 above)
Comment (ignatenkobrain) [inline Dockerfile]: set -x does this

File: Dockerfile (hunk 4 above)
Comment (ignatenkobrain) [inline Dockerfile]: set -x does this

File: Dockerfile (hunk 3 above)
Comment (j-mracek) [inline Dockerfile]: Still not understand why jjb cannot use 
```if ls /rpms/*.rpm 1>/dev/null 2>&1; then dnf -y install /rpms/*.rpm else echo "RPMs files do not exist" >&2; fi \```
and have to use
``` dnf -y install /rpms/*.rpm ```
The meaning is same, but first one report if there is no rpms. With jjb there is another insurance that there are rpms.

**Comment by MichaelMraka** [comment]
```
Step 8 : RUN set -x     && echo "deltarpm=0" >> /etc/dnf/dnf.conf     && dnf -y install httpd vsftpd python2-behave python2-six python-enum34 python2-whichcraft python-jinja2 python2-pexpect     && if [ $type = "local" ] ; then         && dnf -y install dnf-plugins-core python3-dnf-plugins-core python2-dnf-plugins-core rpm-build createrepo_c          && dnf -y copr enable rpmsoftwaremanagement/dnf-nightly ;     fi     && rm -vf /rpms/*extras-versionlock*.rpm /rpms/*extras-local*.rpm /rpms/*extras-torproxy*.rpm     && dnf -y --best upgrade dnf     && if [ $type = "local" ] ; then         if ls /rpms/*.rpm 1>/dev/null 2>&1; then dnf -y install /rpms/*.rpm; fi         && rpm -q dnf         && dnf -y mark install $(dnf repoquery --unneeded -q) ;     else         dnf -y install /rpms/*.rpm         && dnf -y autoremove ;     fi     && dnf -y clean all     && mkdir /tmp/repos.d && mv /etc/yum.repos.d/* /tmp/repos.d/     && mkdir /repo     && rm -f /behave/*.feature
 ---> Running in 8686ffe2cfa1
/bin/sh: -c: line 0: syntax error near unexpected token `&&'
/bin/sh: -c: line 0: `set -x     && echo "deltarpm=0" >> /etc/dnf/dnf.conf     && dnf -y install httpd vsftpd python2-behave python2-six python-enum34 python2-whichcraft python-jinja2 python2-pexpect     && if [ $type = "local" ] ; then         && dnf -y install dnf-plugins-core python3-dnf-plugins-core python2-dnf-plugins-core rpm-build createrepo_c          && dnf -y copr enable rpmsoftwaremanagement/dnf-nightly ;     fi     && rm -vf /rpms/*extras-versionlock*.rpm /rpms/*extras-local*.rpm /rpms/*extras-torproxy*.rpm     && dnf -y --best upgrade dnf     && if [ $type = "local" ] ; then         if ls /rpms/*.rpm 1>/dev/null 2>&1; then dnf -y install /rpms/*.rpm; fi         && rpm -q dnf         && dnf -y mark install $(dnf repoquery --unneeded -q) ;     else         dnf -y install /rpms/*.rpm         && dnf -y autoremove ;     fi     && dnf -y clean all     && mkdir /tmp/repos.d && mv /etc/yum.repos.d/* /tmp/repos.d/     && mkdir /repo     && rm -f /behave/*.feature'
```

**Comment by ignatenkobrain** [comment]
@MichaelMraka fixed syntax errors

**Pull request merged**